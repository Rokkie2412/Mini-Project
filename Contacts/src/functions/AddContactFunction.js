import { launchCamera,launchImageLibrary } from "react-native-image-picker"

export const allowAddTrueFalse = (firstname,lastname,age,disable,setDisable) =>{
    if((firstname === '' || lastname === '') || age === ''){
        setDisable(true)
        return(disable)
    }else{
        setDisable(false)
        return(disable)
    }
}

export const addContactToAPI = (first,last,age,photo) =>{
    const postContact = fetch('https://simple-contact-crud.herokuapp.com/contact',{
        method:'POST',
        body:JSON.stringify({
            firstName : first,
            lastName : last,
            age : age,
            photo : photo
        }),
        headers:{
            'content-type' : 'application/json',
        }
    }).then((Response)=>Response.json())
    .then(json => console.log(json))

    return postContact
}

export const openCamera = (imageFromCamera,setImageFromCamera) => {
        
        const option={
            mediaType : 'photo',
            quality : 0.6,
        }
        launchCamera(option,(res)=>{
            if(res.didCancel){
                console.log('User Cancel Image Picker');
            }else if(res.errorCode){
                console.log(res.errorMessage);
            }else{
                const data = res.assets[0]
                setImageFromCamera(data.uri)
                console.log(data.uri)
            }
        })
        return imageFromCamera
    }

export const openGallery = (imageFromGallery,setImageFromGallery) =>{
    const option = {
        mediaType : 'photo',
        quality : 0.6
    }
    launchImageLibrary(option,(res)=>{
        if(res.didCancel){
            console.log('User Cancel Image Picker')
        }else if(res.errorCode){
            console.log(res.errorCode)
        }else{
            const data= res.assets[0]
            setImageFromGallery(data.uri)
            console.log(data.uri)
        }
    })
    return imageFromGallery
}