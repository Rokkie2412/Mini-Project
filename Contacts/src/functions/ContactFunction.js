export const GettingData = (Contact,setContact,setTempContact) =>{
    fetch('https://simple-contact-crud.herokuapp.com/contact')
    .then((response)=>response.json())
    .then((json)=>{
        setContact(json.data)
        setTempContact(json.data)
    })
    .catch((error)=>console.log(error))
    return Contact
}

export const FilterSearch = (text,Contact,setTempContact,setText) => {
    if(text){
        const newData = Contact.filter((item)=>{
            const itemData = item.firstName ? item.firstName.toUpperCase() : "".toUpperCase()
            const textData = text.toUpperCase()
            return itemData.indexOf(textData) > -1
        })
        setTempContact(newData)
        setText(text)
    }else{
        setTempContact(Contact)
        setText(text)
    }
}