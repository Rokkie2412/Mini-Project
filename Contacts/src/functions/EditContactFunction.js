export const editContactToAPI = (first,last,age,photo,id) =>{
    const putContact = fetch(`https://simple-contact-crud.herokuapp.com/contact/${id}`,{
        method:'PUT',
        body:JSON.stringify({
            firstName : first,
            lastName : last,
            age : age,
            photo : photo
        }),
        headers:{
            'content-type' : 'application/json',
        }
    }).then((Response)=>Response.json())
    .then(json => console.log(json))

    return putContact
}