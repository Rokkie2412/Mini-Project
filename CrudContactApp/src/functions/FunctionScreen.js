import axios from 'axios'
import * as React from 'react'
import NetInfo from "@react-native-community/netinfo";
import RNRestart from 'react-native-restart';
import { launchCamera,launchImageLibrary } from "react-native-image-picker"
import styles from '../styles/ContactAppStyle'
import {View,Text,KeyboardAvoidingView,FlatList,Pressable,Image} from 'react-native'

export const getData = (setData:Function,Data:Object,FilteredData:Object,setFilteredData:Function):Function =>{
    axios.get('https://simple-contact-crud.herokuapp.com/contact')
    .then(res=>{
    setFilteredData(res.data.data)
    setData(res.data.data)
    })
    return Data
}

export const addData = (first:string,last:string,age:number,foto:string):void =>{
    axios.post('https://simple-contact-crud.herokuapp.com/contact',{
        'firstName' : first,
        'lastName' : last,
        'age' : age,
        'photo' : foto
    },
    {
        headers:{
            'content-type' : 'application/json',
        }
    })
}

export const putData = (first:string,last:string,age:number,foto:string,id:string):void =>{
    axios.put(`https://simple-contact-crud.herokuapp.com/contact/${id}`,{
        'firstName' : first,
        'lastName' : last,
        'age' : age,
        'photo' : foto
    },
    {
        headers:{
            'content-type' : 'application/json',
        }
    })
}

export const FilterSearch = (text:string,Contact:Object,setTempContact:Function,setText:Function):void => {
    if(text){
        const newData = Contact.filter((item)=>{
            const itemData = item.firstName ? item.firstName.toUpperCase() : "".toUpperCase()
            const textData = text.toUpperCase()
            return itemData.indexOf(textData) > -1
        })
        setTempContact(newData)
        setText(text)
    }else{
        setTempContact(Contact)
        setText(text)
    }
}

export const openCamera = (image:string,setImage:Function):Function => {
        
        const option={
            mediaType : 'photo',
            quality : 0.6,
        }
        launchCamera(option,(res)=>{
            if(res.didCancel){
                console.log('User Cancel Image Picker');
            }else if(res.errorCode){
                console.log(res.errorMessage);
            }else{
                const data = res.assets[0]
                setImage(data.uri)
                console.log(data.uri)
            }
        })
        return image
    }

export const openGallery = (image:string,setImage:Function):Function =>{
    const option = {
        mediaType : 'photo',
        quality : 0.6
    }
    launchImageLibrary(option,(res)=>{
        if(res.didCancel){
            console.log('User Cancel Image Picker')
        }else if(res.errorCode){
            console.log(res.errorCode)
        }else{
            const data= res.assets[0]
            setImage(data.uri)
            console.log(data.uri)
        }
    })
    return image
}

export const AddorEditContactLock = (first:string,last:string,age:string):Function =>{
    let correctfirstname:boolean = false
    let correctlastname:boolean = false
    let correntage:boolean = false
    let errorText:string = ''
    let allowAdd:boolean = false

    if(first.length >= 3){
        correctfirstname = true
    }else if(first.length <3){
        correctfirstname = false
    }

    if(last.length >= 3){
        correctlastname = true
    }else if(last.length <3){
        correctlastname = false
    }

    if(last.length >= 3){
        correctlastname = true
    }else if(last.length <3){
        correctlastname = false
    }

    if(age.length !== 0 ){
        correntage = true
    }else if(last.length === 0 ){
        correntage = false
    }

    if(correctfirstname === true && correctlastname === true && correntage === true){
        allowAdd = true
    }else{
        allowAdd = false
    }

   return allowAdd
}

export const emptyAll = (setImage:Function,setLast:Function,setFirst:Function,setAge:Function):void =>{
    setImage('N/A')
    setLast('')
    setFirst('')
    setAge('')
}

export const FuncRepeat = (text:string,setRepeat:Function):void =>{
    if(text.length === 0){
        setRepeat(1000)
    }else{
        setRepeat(100000)
    }
}

export const CheckInternet = (set:Function):void =>{
    NetInfo.fetch('https://simple-contact-crud.herokuapp.com/contact')
    .then(networkState=>{
        if(networkState.isConnected !== true){
            set(true)
        }
    })
}
export const AddPack = (setmodal:Function,age:string,first,last,image,setImage,setLast,setFirst,setAge,setError):void =>{
        setmodal(false)
        const tempPasre = parseInt(age)
        addData(first,last,tempPasre,image)
        setError('')
        emptyAll(setImage,setLast,setFirst,setAge)
    }

export const Restart = ():void =>{
    RNRestart.Restart()
}
       