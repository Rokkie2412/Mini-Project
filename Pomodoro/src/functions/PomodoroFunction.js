import reactNativeBackgroundTimer from 'react-native-background-timer';

export const displayTimer = (secondLeft) =>{
    let hours = Math.floor(secondLeft/60/60)
    let minutes = Math.floor(secondLeft/60%60)
    let seconds = Math.floor(secondLeft%60)

    let displayHours = hours < 10 ? `0${hours}` : hours
    let displayMins = minutes < 10 ? `0${minutes}` : minutes
    let displaySecs = seconds < 10 ? `0${seconds}` : seconds

    return{
      displayHours,
      displayMins,
      displaySecs
    }
}

export const startTimer = (setSecondLeft) =>{
  reactNativeBackgroundTimer.runBackgroundTimer(()=>{
    setSecondLeft(secs =>{
      if(secs > 0){
        return secs-1
      }else if (secs === 0 ) {
        reactNativeBackgroundTimer.stopBackgroundTimer()
      }
    })
  },1000)
}

export const onchecklimit = (value,setValue) =>{
    const parsedqty = Number.parseInt(value)
    if(Number.isNaN(parsedqty))
      setValue(0)
    else if( parsedqty >59)
      setValue(59)
    else
      setValue(parsedqty)
  }