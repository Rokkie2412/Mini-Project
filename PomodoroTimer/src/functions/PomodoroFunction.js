import reactNativeBackgroundTimer from 'react-native-background-timer';
import PushNotification from "react-native-push-notification";
import { Vibration } from 'react-native';
export const displayTimer = (secondLeft) =>{
    let hours = Math.floor(secondLeft/60/60)
    let minutes = Math.floor(secondLeft/60%60)
    let seconds = Math.floor(secondLeft%60)

    let displayHours = hours < 10 ? `0${hours}` : hours
    let displayMins = minutes < 10 ? `0${minutes}` : minutes
    let displaySecs = seconds < 10 ? `0${seconds}` : seconds

    return{
      displayHours,
      displayMins,
      displaySecs
    }
}

export const startTimer = (setSecondLeft) =>{
  reactNativeBackgroundTimer.runBackgroundTimer(()=>{
    setSecondLeft(secs =>{
      if(secs > 0){
        return secs-1
      }else if (secs === 0 ) {
        reactNativeBackgroundTimer.stopBackgroundTimer()
      }
    })
  },1000)
}

export const onchecklimit = (value,setValue) =>{
    const parsedqty = Number.parseInt(value)
    if(Number.isNaN(parsedqty))
      setValue(0)
    else if( parsedqty >59)
      setValue(59)
    else
      setValue(parsedqty)
  }

export const createChannel = () =>{
    PushNotification.createChannel({
      channelId:'test-channel',
      channelName:'Test Channel',
      vibrate:true
    })
  }

export const handleNotification = (judul,pesan) =>{
    PushNotification.localNotification({
      channelId:"test-channel",
      title:judul,
      message:pesan,
      playSound:true,
      id:1,
      vibrate:true,
      vibration:1000
    })
  }
  
export const handleNotificationLongBreak = () =>{
    PushNotification.localNotification({
      channelId:"test-channel",
      title:'Time to long break!',
      message:"it's time to long break! Use it wisely",
      playSound:true,
      id:1,
      vibrate:true,
      vibration:1000,
    })
  }

export const TimerinNotification = (judul,pesan) =>{
    PushNotification.localNotification({
      channelId:"test-channel",
      title:judul,
      message:pesan,
      playSound:false,
      id:2,
      vibrate:false,
    })
  }