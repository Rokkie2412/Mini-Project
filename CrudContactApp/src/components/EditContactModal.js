/**
 * @format
 * @flow
 */
import * as React from 'react'
import {useState,useEffect} from 'react'
import {View,KeyboardAvoidingView,Pressable,TextInput,Text,Image} from 'react-native'
import Modal from "react-native-modal";
import type {Add} from './component.type'
import {putData,AddorEditContactLock} from '../functions/FunctionScreen'
import Icon from 'react-native-vector-icons/Ionicons'
import styles from '../styles/EditContact.style'
import type {EditContact} from './component.type'
import AwesomeAlert from './AwesomeAlert';


const EditContactModal = 
({show,setshow,id,first,last,age,photo,setfirst,setlast,setage,setphoto}:EditContact):React.Node =>{
    const [pickImage,setPickImage] = useState(false)
    const [Error,setError] = useState('')

    const putPack = ():void =>{
        setshow(false)
        const tempPasre = parseInt(age)
        putData(first,last,tempPasre,photo,id)
        setError('')
    }

    const set = ():void =>{
        AddorEditContactLock(first,last,age) === true?
        putPack():
        setError('*Make sure name have 3 character length  and age between 1-99')
    }

    return(
        <KeyboardAvoidingView>
            <Modal
            isVisible={show}
            style={styles.mainContainer}
            >
            <Text style={styles.header}>Edit Contact</Text>
            <Pressable 
            testID='PickImage'
            onPress={()=>setPickImage(true)}
            style={styles.imageView}>
                {photo === "N/A" ? <Icon style={styles.personIcon} name='person'/>:
                <Image style={styles.imagePerson} source={{uri:`${photo}`}}/>
                }
            </Pressable>
            <View style={styles.inputsection}>
                <TextInput
                testID='first'
                placeholderTextColor="#eeeeee"
                placeholder='Edit First Name'
                style={styles.textinput}
                value={first}
                onChangeText={(temp)=>setfirst(temp)}
                />
                <TextInput
                testID='last'
                placeholderTextColor="#eeeeee"
                placeholder='Edit Last Name'
                style={styles.textinput}
                value={last}
                onChangeText={(temp)=>setlast(temp)}
                />
                <TextInput
                testID='age'
                placeholderTextColor="#eeeeee"
                placeholder='Edit Age'
                style={styles.textinputAge}
                value={age.toString()}
                onChangeText={(temp)=>setage(temp)}
                />
            </View>
            <Text style={styles.error}>{Error}</Text>
            <View style={styles.buttonSection}>
                <Pressable
                testID='cancel'
                onPress={()=>{
                    setshow(false)
                }}
                >
                    <Text style={styles.button}>Cancel</Text>
                </Pressable>
                <Pressable
                testID='edit'
                onPress={()=>{
                   set()
                }}
                >
                    <Text style={styles.button}> Edit </Text>
                </Pressable>
            </View>
            </Modal>
            <AwesomeAlert show={pickImage} setshow={setPickImage} image={photo} setimage={setphoto}/>
        </KeyboardAvoidingView>
    )
}

export default EditContactModal