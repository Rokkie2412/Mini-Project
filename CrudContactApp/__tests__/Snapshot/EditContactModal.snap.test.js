import {create,act} from 'react-test-renderer';
import React from 'react';
import EditContactModal from "../../src/components/EditContactModal";
import {AddorEditContactLock} from "../../src/functions/FunctionScreen"
const func = jest.fn()
const tree = create(<EditContactModal photo={'N/A'} setshow={func} first={"kota"} last={"Jambi"} age={"12"} setfirst={func} setlast={func} setage={func} setphoto={func}/>)

jest.mock('react-native-vector-icons/Ionicons', () => 'Mocked Ionicons');
jest.mock('react-native-modal',()=> "Mocked Modal React Native")
jest.mock('@react-native-community/netinfo',()=>'Mocked Netinfo')
jest.mock('react-native-image-picker',()=>{
    return{
        launchCamera:jest.fn(),
        launchImageLibrary:jest.fn()
    }
})

// const spyHandleNotification = jest.spyOn(age,'toString') 

describe('Testing for Modal Edit Contact', () => {
  describe('Snapshot for Edit Contact', () => {
    it("Should render correctly",()=>{
        expect(tree).toMatchSnapshot()
    })
  })
  describe('Button and Textinput test for Edit Contact', () => {
    it("Should be pressed",()=>{
        const pick = tree.root.findByProps({testID:'PickImage'}).props
        act(()=>pick.onPress())
        const edit = tree.root.findByProps({testID:'edit'}).props
        act(()=>edit.onPress())
        
        const cancel = tree.root.findByProps({testID:'cancel'}).props
        act(()=>cancel.onPress())
        // expect(AddorEditContactLock("kota","Jambi","12")).toBeCalled(true)
    })
    it("Should correct onChangeText",()=>{
        const first = tree.root.findByProps({testID:'first'}).props
        act(()=>first.onChangeText())
        const last = tree.root.findByProps({testID:'last'}).props
        act(()=>last.onChangeText())
        const age = tree.root.findByProps({testID:'age'}).props
        act(()=>age.onChangeText())
    })
  })
  
})
