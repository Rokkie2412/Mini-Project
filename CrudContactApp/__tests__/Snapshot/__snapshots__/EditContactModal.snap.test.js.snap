// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for Modal Edit Contact Snapshot for Edit Contact Should render correctly 1`] = `
<View
  onLayout={[Function]}
>
  <Mocked Modal React Native
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#393E46",
        "borderRadius": 30,
        "flex": 1,
        "height": "67%",
        "justifyContent": "center",
        "marginHorizontal": "10%",
        "marginVertical": "25%",
        "position": "absolute",
        "width": "95%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#EEEEEE",
          "fontSize": 32,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Edit Contact
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={Object {}}
      testID="PickImage"
    >
      <Mocked Ionicons
        name="person"
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#00ADB5",
            "borderRadius": 50,
            "color": "#EEEEEE",
            "fontSize": 50,
            "margin": "9%",
            "padding": 15,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "marginHorizontal": "14%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Edit First Name"
        placeholderTextColor="#eeeeee"
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#eeeeee",
            "color": "#EEEEEE",
            "marginBottom": "7%",
          }
        }
        testID="first"
        value="kota"
      />
      <TextInput
        onChangeText={[Function]}
        placeholder="Edit Last Name"
        placeholderTextColor="#eeeeee"
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#eeeeee",
            "color": "#EEEEEE",
            "marginBottom": "7%",
          }
        }
        testID="last"
        value="Jambi"
      />
      <TextInput
        onChangeText={[Function]}
        placeholder="Edit Age"
        placeholderTextColor="#eeeeee"
        style={
          Object {
            "alignSelf": "center",
            "borderBottomWidth": 1,
            "borderColor": "#eeeeee",
            "color": "#EEEEEE",
            "marginBottom": "7%",
            "textAlign": "center",
            "width": "20%",
          }
        }
        testID="age"
        value="12"
      />
    </View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#EEEEEE",
          "fontSize": 7,
        }
      }
    >
      
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="cancel"
      >
        <Text
          style={
            Object {
              "backgroundColor": "#00ADB5",
              "borderRadius": 50,
              "color": "#EEEEEE",
              "fontSize": 18,
              "fontWeight": "bold",
              "padding": 15,
            }
          }
        >
          Cancel
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="edit"
      >
        <Text
          style={
            Object {
              "backgroundColor": "#00ADB5",
              "borderRadius": 50,
              "color": "#EEEEEE",
              "fontSize": 18,
              "fontWeight": "bold",
              "padding": 15,
            }
          }
        >
           Edit 
        </Text>
      </View>
    </View>
  </Mocked Modal React Native>
  <Mocked Modal React Native
    isVisible={false}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#EEEEEE",
        "borderRadius": 30,
        "flex": 1,
        "height": "10%",
        "marginVertical": "50%",
        "position": "absolute",
        "width": "60%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="Close"
    >
      <Mocked Ionicons
        name="close"
        style={
          Object {
            "alignSelf": "flex-end",
            "backgroundColor": "#222831",
            "borderRadius": 50,
            "color": "#EEEEEE",
            "marginBottom": 20,
            "marginRight": 10,
            "padding": 8,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#222831",
          "fontSize": 18,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Choose image source :
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="openCamera"
      >
        <Mocked Ionicons
          name="camera"
          style={
            Object {
              "backgroundColor": "#222831",
              "borderRadius": 50,
              "color": "#EEEEEE",
              "fontSize": 24,
              "margin": 10,
              "padding": "10%",
            }
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="openGallery"
      >
        <Mocked Ionicons
          name="image"
          style={
            Object {
              "backgroundColor": "#222831",
              "borderRadius": 50,
              "color": "#EEEEEE",
              "fontSize": 24,
              "margin": 10,
              "padding": "10%",
            }
          }
        />
      </View>
    </View>
  </Mocked Modal React Native>
</View>
`;
